peer/network.go

```diff
@@ -505,12 +505,11 @@ func (n *network) Connected(ctx context.Context, nodeID ids.NodeID, nodeVersion
 		return nil
 	}
 
-	if nodeID == n.self {
+	if nodeID != n.self {
 		log.Debug("skipping registering self as peer")
-		return nil
+		n.peers.Connected(nodeID, nodeVersion)
 	}
```

Comment:
I had the same idea initially, but having `nodeID != n.self` and logging `skipping registering self as peer` might be confusing. I ended up with this.
```go
if nodeID != n.self {
    n.peers.Connected(nodeID, nodeVersion)
} else {
    log.Debug("skipping registering self as peer")
}
```

Reply:
I think there is some work to be done in the p2p sdk to standardize how it handles the registration of itself as a peer. So this PR is likely to change a bit.

Reply:
But yeah as this PR is currently written that log is incorrect.